comm_sbs_bqctrl.py

```diff
@@ -3649,26 +3653,38 @@ def main():
             print("Opening {}".format(po.bus))
         smbus_open(po.bus, po)
 
+    # Re-init global variables; then they are modified by specific chip driver, do it before detection
+    reset_default_driver(po)
     if po.chip == CHIP_TYPE.AUTO:
         po.chip = smart_battery_detect(vals, po)
 
-    # Re-init global variables; then they are modified by specific chip driver
-    reset_default_driver(po)
-    if po.chip in (CHIP_TYPE.BQ30z50, CHIP_TYPE.BQ30z55, CHIP_TYPE.BQ30z554,):
-        fnames = ["comm_sbs_chips/{}.py".format("BQ30z554")]
-    elif po.chip in (CHIP_TYPE.BQ40z50,):
-        fnames = ["comm_sbs_chips/{}.py".format("BQ40z50")]
+    if(po.chip in driver_cache):
+        chip_file_code = driver_cache[po.chip]
+        exec(chip_file_code)
     else:
-        fnames = ["comm_sbs_chips/{}.py".format(po.chip.name)]
-    for fname in fnames:
-        try:
-            with open(fname, "rb") as source_file:
-                chip_file_code = compile(source_file.read(), fname, "exec")
-            if (po.verbose > 0):
-                print("Importing {}".format(fname))
-            exec(chip_file_code)
-        except IOError:
-            print("Warning: Could not open chip definition file '{}'".format(fname))
+        if po.chip in (CHIP_TYPE.BQ30z50, CHIP_TYPE.BQ30z55, CHIP_TYPE.BQ30z554,):
+            fnames = ["comm_sbs_chips/{}.py".format("BQ30z554")]
+        elif po.chip in (CHIP_TYPE.BQ40z50,):
+            fnames = ["comm_sbs_chips/{}.py".format("BQ40z50")]
+        elif po.chip in ("SBS",):    # default
+            pass # do nothing, already loaded with reset_default_driver(po)
+        else:
+            fnames = ["comm_sbs_chips/{}.py".format(po.chip.name)]
+        driver_cache[po.chip] = list()
+        for fname in fnames:
+            try:
+                with open(fname, "rb") as source_file:
+                    chip_file_code = compile(source_file.read(), fname, "exec")
+                if (po.verbose > 0):
+                    print("Importing {}".format(fname))
+                exec(chip_file_code)
+                driver_cache[po.chip] = chip_file_code
+            except IOError:
+                print("Warning: Could not open chip definition file '{}'".format(fname))
+            # end try
```

Comment:
In the coding style enforced by the script, there are no ending comments. Please remove.

Reply:
fixed